### PR TITLE
Add layer between UI and core for returns

### PR DESCRIPTION
**Pull request by danielburnley**
WHAT

Adds usecases between the UI and the core application for returns

WHY

First steps towards allowing us to update the frontend schema without having to change the way it is internally stored

### CODE REVIEW

**Comment by danielburnley** [comment]
Depends on #152

File: spec/unit/ui/use_case/create_project_spec.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        have_received(:execute).with(hash_including(baseline: { Cats: 'purr' }))
+      )
+    end
+
+    it 'Returns the id from create project' do
+      expect(response).to eq(id: 1)
+    end
+  end
+
+  context 'Example one' do
```
Comment (ccp92) [inline spec/unit/ui/use_case/create_project_spec.rb]:
```suggestion
  context 'Example two' do
```
You've put example one twice.

File: spec/unit/ui/use_case/update_project_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        have_received(:execute).with(
+          hash_including(
+            project_data: { dog: 'woof' }
+          )
+        )
+      )
+    end
+
+    it 'Returns successful if successful' do
+      expect(response).to eq(successful: false)
```
Comment (ccp92) [inline spec/unit/ui/use_case/update_project_spec.rb]: This test is a little confusing?
It should return successful but we're expecting `successful: false`?
(Sorry, I'm probably missing something obvious)

File: spec/unit/ui/use_case/update_return_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        )
+      )
+    end
+
+    it 'returns an empty hash' do
+      expect(response).to eq({})
+    end
+  end
+
+  context 'Example one' do
```
Comment (ccp92) [inline spec/unit/ui/use_case/update_return_spec.rb]:
```suggestion
  context 'Example two' do
```

File: spec/unit/ui/use_case/update_project_spec.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      expect(update_project_spy).to(
+        have_received(:execute).with(
+          hash_including(
+            project_data: { dog: 'woof' }
+          )
+        )
+      )
+    end
+
+    it 'Returns successful if successful' do
```
Comment (danielburnley) [inline spec/unit/ui/use_case/update_project_spec.rb]:
```suggestion
    it 'Returns unsuccessful if unsuccessful' do
```

File: spec/unit/ui/use_case/update_project_spec.rb (hunk 2 above)
Comment (danielburnley) [reply spec/unit/ui/use_case/update_project_spec.rb]: Badly worded test - have updated

**Pull request merged**